src/VecSim/algorithms/brute_force/brute_force.cpp

```diff
@@ -330,14 +329,13 @@ VecSimBatchIterator *BruteForceIndex::newBatchIterator(const void *queryBlob,
         float_vector_normalize((float *)queryBlobCopy, dim);
     }
     // Ownership of queryBlobCopy moves to BF_BatchIterator that will free it at the end.
-    return new (this->allocator)
-        BF_BatchIterator(queryBlobCopy, this, queryParams, this->allocator);
+    return newBatchIterator_Instance(queryBlobCopy, queryParams);
 }
 
 bool BruteForceIndex::preferAdHocSearch(size_t subsetSize, size_t k, bool initial_check) {
     // This heuristic is based on sklearn decision tree classifier (with 10 leaves nodes) -
     // see scripts/BF_batches_clf.py
-    size_t index_size = this->indexSize();
+    size_t index_size = this->indexLabelCount(); // Index's unique label size
```

Comment:
I'm not sure that it's the right thing to do. We still want to evaluate to ratio between the `subsetSize` to the actual `indexSize`, no?